src/napari_imagej/java.py

```diff
@@ -2,118 +2,207 @@
+    """
+    init_ij_async()
+    initializer.wait()
+
+
+def init_ij_async():
+    """
+    Starts creating our ImageJ instance.
+    """
+    initializer.start()
+
+
 def ij():
     """
     Returns the ImageJ instance.
     If it isn't ready yet, blocks until it is ready.
     """
-    return ij_init().get()
+    init_ij()
+    return initializer.ij
 
 
-def ensure_jvm_started() -> None:
+class ImageJInitializer(QThread):
     """
-    Blocks until the ImageJ instance is ready.
+    A QThread responsible for initializing ImageJ
     """
-    ij_init().wait()
 
+    # Tools used to ensure this QThread only initializes ImageJ once.
+    started: bool = False
+    lock: Lock = Lock()
+    # Used as a placeholder for the ImageJ instance
+    ij = None
 
-def _imagej_init():
-    # Initialize ImageJ
-    log_debug("Initializing ImageJ2")
+    def run(self):
+        """
+        Main QThread loop.
 
-    # -- IMAGEJ CONFIG -- #
+        Well, okay, it isn't really a loop.
 
-    # TEMP: Avoid issues caused by
-    # https://github.com/imagej/pyimagej/issues/160
-    config.add_repositories(
-        {"scijava.public": "https://maven.scijava.org/content/groups/public"}
-    )
-    config.add_option(f"-Dimagej2.dir={settings['imagej_base_directory'].get(str)}")
-    log_debug("Completed JVM Configuration")
+        Responsible for ensuring that initialize is only called once.
+        """
+        # Double-checked lock
+        if not self.started:
+            with self.lock:
+                if not self.started:
+                    self.started = True
+                    # Try to initialize ImageJ
+                    try:
+                        self.initialize()
+                        # Report completion
+                        java_signals._startup_complete.emit()
+                    # We failed!
+                    except Exception as exc:
+                        # Report failure
+                        java_signals._startup_error.emit(exc)
+
+    def initialize(self):
+        """
+        Creates the ImageJ instance
+        """
+        # Initialize ImageJ
+        log_debug("Initializing ImageJ2")
+
+        # -- IMAGEJ CONFIG -- #
+
+        # TEMP: Avoid issues caused by
+        # https://github.com/imagej/pyimagej/issues/160
+        config.add_repositories(
+            {"scijava.public": "https://maven.scijava.org/content/groups/public"}
+        )
+        config.add_option(f"-Dimagej2.dir={settings['imagej_base_directory'].get(str)}")
+        log_debug("Completed JVM Configuration")
+
+        # Add converters
+        from napari_imagej.types.converters import install_converters
+
+        install_converters()
+
+        # Launch PyImageJ
+        self.ij = imagej.init(
+            ij_dir_or_version_or_endpoint=settings["imagej_directory_or_endpoint"].get(
+                str
+            ),
+            mode=settings["jvm_mode"].get(str),
+            add_legacy=settings["include_imagej_legacy"].get(bool),
+        )
```

Comment:
We could consider moving the configuration parts done within `imagej.init` up to the config section

Reply:
Fixed in 076b8ad